modules/exploits/windows/local/ms15_051_client_copy_image.rb

```diff
@@ -73,11 +73,7 @@ def check
       return Exploit::CheckCode::Unknown
     end
 
-    if sysinfo['Architecture'] =~ /(wow|x)64/i
-      arch = ARCH_X64
-    elsif sysinfo['Architecture'] =~ /x86/i
-      arch = ARCH_X86
-    end
+    arch = sysinfo['Architecture']
```

Comment:
This actually doesn't look like it's used anywhere. Are we sure this is still necessary? I'm wondering if the original had a fail on wow64 condition that was removed for the purposes of the check method.

Reply:
This doesn't look to be necessary in the current code. Previous code I see this definition, which still doesn't make much sense: as `arch` isn't being used in the code, but it does show the old `wow64` condition check.

```
  def check
    # Windows XP SP3 (32-bit)                      5.1.2600.6514 (Works)
    # Windows Server 2003 Standard SP2 (32-bit)    5.2.3790.5445 (Works)
    # Windows Server 2008 Enterprise SP2 (32-bit)  6.0.6002.18005 (Does not work)
    # Windows 7 SP1 (64-bit)                       6.1.7601.17514 (Works)
    # Windows 7 SP1 (64-bit)                       6.1.7601.17535 (Works)
    # Windows 7 SP1 (32-bit)                       6.1.7601.17514 (Works)
    # Windows 7 SP1 (32-bit)                       6.1.7601.18388 (Works)
    # Windows Server 2008 R2 (64-bit) SP1          6.1.7601.17514 (Works)
    # Windows Server 2008 R2 (64-bit) SP1          6.1.7601.18105 (Works)
    if sysinfo['OS'] !~ /windows/i
      return Exploit::CheckCode::Unknown
    end
    if sysinfo['Architecture'] =~ /(wow|x)64/i
      arch = ARCH_X64
    elsif sysinfo['Architecture'] =~ /x86/i
      arch = ARCH_X86
    end
    file_path = expand_path('%windir%') << '\\system32\\win32k.sys'
    major, minor, build, revision, branch = file_version(file_path)
    vprint_status("win32k.sys file version: #{major}.#{minor}.#{build}.#{revision} branch: #{branch}")
    return Exploit::CheckCode::Safe if build > 7601
    return Exploit::CheckCode::Appears
  end
```

Reply:
Given the above and the lack of a use case here it may be better to just remove this code as mentioned. I'll make an update now to fix this.

Reply:
Should be resolved with 41794fe now.